traefik/values.yaml

```diff
@@ -266,7 +266,13 @@ metrics:
   #    address: localhost:8125
 ##
 ##  enable optional CRDs for Prometheus Operator
-##
+## 
+    # create a dedicated metrics service for use with servicemonitor
+    # will only be created, when enabled set to true and hub.enabled is false
+    service:
+      enabled: false
```

Comment:
This should be commented and indented following [GuideLines](https://github.com/traefik/traefik-helm-chart/blob/master/traefik/Guidelines.md#optional-non-critical-feature-example)

Reply:
Ok - i did. I used a different solution for the nested nil issue
```
{{- if (and (.Values.metrics.prometheus.service).enabled (not .Values.hub.enabled)) -}}
```
instead of 
```
{{- if (and .Values.metrics.prometheus.service -}}
{{- if (and .Values.metrics.prometheus.service.enabled (not .Values.hub.enabled)) -}}
```

I did so, because otherwise the if-clause in the servicemonitor labels would become overly complex. Hope that's fine for you.